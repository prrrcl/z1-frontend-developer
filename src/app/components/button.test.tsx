import React from 'react'
import renderer from 'react-test-renderer'
import Button from './button'

describe('Button', () => {
  const children = 'Button'
  const onClick = () => console.log('Button')

  it('[Props] Check props', () => {
    expect(children).toMatchSnapshot(expect.any(String))
  })

  it('[Render] Button without props', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot()
  })

  it("[Render] Button with Props", () => {
    const tree = renderer.create(
      <Button onClick={onClick}>
        {children}
      </Button>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})