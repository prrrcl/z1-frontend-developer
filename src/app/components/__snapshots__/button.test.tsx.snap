// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button [Props] Check props 1`] = `
Object {
  "0": "B",
  "1": "u",
  "2": "t",
  "3": "t",
  "4": "o",
  "5": "n",
}
`;

exports[`Button [Render] Button with Props 1`] = `
<button
  className="css-al3l4-Button"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button [Render] Button without props 1`] = `
<button
  className="css-al3l4-Button"
  onClick={[Function]}
>
  Button
</button>
`;
